### PR TITLE
Clean-up, cmd lint resolve, refactor and short-term fixing DPS IT.

### PR DESCRIPTION
**Pull request by digimaun**
- The generic show command ensures a consistent behavior when encountering a missing Azure resource (like throwing an exit code of 3 if resource not found)

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [x] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [x] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [x] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.
- [x] Have you made an entry in HISTORY.rst which concisely explains your feature or change?

### CODE REVIEW

**Comment by digimaun** [comment]
Single failure from TestCLIInit caused by an unclean state of my branch

image

Cleaned up branch state

image

File: azext_iot/tests/test_iot_central_int.py [hunk 1]
```diff
@@ -197,7 +197,7 @@ def test_device_connect(self):
         device_id = "testDevice"
 
         device_primary_key = self.cmd(
-            "iot central device compute-device-key create --pk {} -d {}".format(
+            "iot central device compute-device-key --pk {} -d {}".format(
```
Comment (prbans) [inline azext_iot/tests/test_iot_central_int.py:200]: @valluriraj does this align with PM ask?

File: azext_iot/tests/test_iot_dps_int.py [hunk 2]
```diff
@@ -20,6 +20,11 @@
 
 cert_name = 'test'
 cert_path = cert_name + '-cert.pem'
+test_endorsement_key = ('AToAAQALAAMAsgAgg3GXZ0SEs/gakMyNRqXXJP1S124GUgtk8qHaGzMUaaoABgCAAEMAEAgAAAAAAAEAibym9HQP9vxCGF5dVc1Q'
```
Comment (prbans) [inline azext_iot/tests/test_iot_dps_int.py:23]: do we want keys in plaintext like this?

Comment (digimaun) [reply azext_iot/tests/test_iot_dps_int.py:23]: It's completely made up

File: azext_iot/tests/test_iot_central_int.py (hunk 1 above)
Comment (valluriraj) [reply azext_iot/tests/test_iot_central_int.py:200]: synced up with PM good to go.

File: azext_iot/tests/test_iot_dps_int.py (hunk 2 above)
Comment (digimaun) [reply azext_iot/tests/test_iot_dps_int.py:23]: This is a sample TPM public key.

File: azext_iot/product/test/command_map.py [hunk 3]
```diff
@@ -26,15 +26,15 @@ def load_product_test_commands(self, _):
...
+        g.show_command("show", "show")
         g.command("search", "search")
     with self.command_group("iot product test case", command_type=test_cases_ops) as g:
         g.command("list", "list")
         g.command("update", "update")
     with self.command_group("iot product test task", command_type=test_tasks_ops) as g:
         g.command("create", "create")
         g.command("delete", "delete")
-        g.command("show", "show")
+        g.show_command("show", "show")
```
Comment (prbans) [inline azext_iot/product/test/command_map.py:37]: whats the purpose of this show command btw

Comment (digimaun) [reply azext_iot/product/test/command_map.py:37]: There was a new command table custom linter rule added as of a couple days ago. It's asking for `show` commands to use `show_command` to ensure consistent behavior when encountering a missing Azure resource (like throwing an exit code of 3 if resource not found).

**Pull request merged**